### PR TITLE
Refactor tree display code

### PR DESCRIPTION
**Pull request by KSchopmeyer**
Refactor the code in the _displaytree.py module to make it more comprehensible, faster, and correct an outstanding issue where we were not always displaying the tree in the same order and ignoreing possible issues of classname case-independence

This also adds tests for boundary conditions of the "class tree" command

This pr does not have a defined issue since it does not change external behavior.

It was tested against both 0.17 and 1.0.0 b2 and handles the full 400+ class hiearchy tree display in about 4 seconds.

### CODE REVIEW

File: pywbemtools/pywbemcli/_cmd_class.py [hunk 1]
```diff
@@ -823,58 +823,79 @@ def cmd_class_find(context, classname_glob, options):
...
-    left-justified tree using the asciitree library.
-    The --superclasses option determines if the superclass tree or the
-    subclass tree is displayed.
+    Get the class hiearchy from the server, either the superclasses
+    associated with a class or the subclasses.  If getting subclasses
+    the classname parameter may be None which requests that the complete
+    class hiearchy be retrieved..
+
+      Parameters:
+        conn(TODO):
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/_cmd_class.py:834]: Change to:
```
    conn(WBEMConnection): Connection to the server.
```

File: pywbemtools/pywbemcli/_cmd_class.py [hunk 2]
```diff
@@ -823,58 +823,79 @@ def cmd_class_find(context, classname_glob, options):
...
+
+        classname(:term:`string):
+            classname if the tree is to be initiated from
+            within the class hiearchy. May be None
+
+        namespace(:term:`string):
+          Namespace to use to acquire classes from WBEM server
+
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/_cmd_class.py:842]: Add a description of the `superclasses` parameter.

File: pywbemtools/pywbemcli/_cmd_class.py [hunk 3]
```diff
@@ -823,58 +823,79 @@ def cmd_class_find(context, classname_glob, options):
...
+        CIM_Error:
     """
 
     # TODO FUTURE: Sort out how we handle output format with tree output.
     try:
-        if options['superclasses']:
+        if superclasses:
             if classname is None:
                 raise click.ClickException('CLASSNAME argument required for '
                                            '--superclasses option')
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/_cmd_class.py:854]: It seems to me that such a handling of missing command arguments should be done by the caller, and not in this retrieval function.

File: pywbemtools/pywbemcli/_cmd_class.py [hunk 4]
```diff
@@ -823,58 +823,79 @@ def cmd_class_find(context, classname_glob, options):
...
+        conn(TODO):
+
+        classname(:term:`string):
+            classname if the tree is to be initiated from
+            within the class hiearchy. May be None
+
+        namespace(:term:`string):
+          Namespace to use to acquire classes from WBEM server
+
+      Returns:
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/_cmd_class.py:843]: Add description of return value

File: pywbemtools/pywbemcli/_cmd_class.py (hunk 1 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_class.py:834]: DONE

File: pywbemtools/pywbemcli/_cmd_class.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_class.py:842]: Done

File: pywbemtools/pywbemcli/_cmd_class.py (hunk 4 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_class.py:843]: DONE

File: pywbemtools/pywbemcli/_cmd_class.py (hunk 3 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_class.py:854]: Do not understand this comment.  The code is

if superclasses:

It is not there as an assertion but as a test whether the user wanted superclasses or subclasses.

**DISCUSSION**

Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_class.py:861]: That was also a good change, but I meant the whole check on lines 858 to 861 (in the changed version):
```
        if superclasses:
            if classname is None:
                raise click.ClickException('CLASSNAME argument required for '
                                           '--superclasses option')
```
That check is already done in cmd_class_tree() which is the only caller.
So the check here could be changed to become an assertion:
```
        if superclasses:
            assert classname is not None
```

**Comment by andy-maier** [comment]
There is one comment left - see above.

File: pywbemtools/pywbemcli/_cmd_class.py (hunk 3 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_class.py:861]: Whoops, missed that duplication.  Fixed. DONE

File: pywbemtools/pywbemcli/_cmd_class.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_class.py:843]: DONE
